src/client/datascience/notebook/kernelWithMetadata.ts

```diff
@@ -71,7 +78,7 @@ export class VSCodeNotebookKernelMetadata implements VSCNotebookKernel {
                     ranges.some((range) => range.start <= cell.index && cell.index < range.end)
             );
         traceInfo(`Execute Cells request ${cells.length} ${cells.map((cell) => cell.index).join(', ')}`);
-        await cells.map((cell) => this.executeCell(document, cell));
+        await Promise.all(cells.map((cell) => this.executeCell(document, cell)));
```

Comment:
I'm not sure about this change, but the original code we had here looked a little odd to me. We were awaiting on an array of promises, which TypeScript flagged as a no-op:
![image](https://user-images.githubusercontent.com/30305945/114122631-8991ed80-98a5-11eb-9510-25c117202158.png)

I think the intent was to await on Promise.all of that array so I went ahead and made the change.

Reply:
Yikes, good find.